carma-messenger-core/cpp_message/src/MobilityRequest_Message.cpp

```diff
@@ -0,0 +1,538 @@
+            {
+                for(size_t i=0;i<str_len;i++){
+                    sender_id +=message->value.choice.TestMessage00.header.hostStaticId.buf[i];
+                }
+            }
+            else sender_id=Header_constant.STRING_DEFAULT;
+
+            header.sender_id=sender_id;
+
+            //get recepient id
+            str_len=message->value.choice.TestMessage00.header.targetStaticId.size;
+            if(str_len<=Header_constant.STATIC_ID_MAX_LENGTH && str_len!=0)
+            {
+                for(size_t i=0;i<str_len;i++){
+                    recipient_id +=message->value.choice.TestMessage00.header.targetStaticId.buf[i];
+                }
+            }
+            else recipient_id=Header_constant.STRING_DEFAULT;
+
+            header.recipient_id=recipient_id;
+            
+            //get bsm id
+            str_len=message->value.choice.TestMessage00.header.hostBSMId.size;
+            if(str_len==Header_constant.BSM_ID_LENGTH)
+            {
+                for(size_t i=0;i<str_len;i++){
+                    sender_bsm_id +=message->value.choice.TestMessage00.header.hostBSMId.buf[i];
+                }
+            }
+            else sender_bsm_id=Header_constant.BSM_ID_DEFAULT;
+            
+            header.sender_bsm_id=sender_bsm_id;
+
+            //get plan id
+            str_len=message->value.choice.TestMessage00.header.planId.size;
+            if(str_len==Header_constant.GUID_LENGTH)
+            {
+                for(size_t i=0;i<str_len;i++){
+                    plan_id +=message->value.choice.TestMessage00.header.planId.buf[i];
+                }
+            }
+            else plan_id=Header_constant.GUID_DEFAULT;
+
+            header.plan_id=plan_id;
+            
+            //recover uint64_t timestamp from string
+            str_len=message->value.choice.TestMessage00.header.timestamp.size;
+            timestamp=0;
+            for(size_t i=0;i<str_len;i++){
+                timestamp*=10;
+                timestamp+=int(message->value.choice.TestMessage00.header.timestamp.buf[i])-'0';
+            }
+            header.timestamp=timestamp;
+
+            output.header=header;
+
+            //strategy
+            std::string strategy;
+            str_len=message->value.choice.TestMessage00.body.strategy.size;
+            if(str_len<=STRATEGY_MAX_LENGTH && str_len!=0)
+            {
+                for(size_t i=0;i<str_len;i++){
+                    strategy +=message->value.choice.TestMessage00.body.strategy.buf[i];
+                }
+            }
+            else strategy="";
+
+            if(strategy==STRING_DEFAULT){
+                strategy="";
+            }
+            output.strategy=strategy;
+            //plan type   
+            MobilityPlanType_t type;
+            type=message->value.choice.TestMessage00.body.planType;
+            enum plan {UNKNOWN=0,CHANGE_LANE_LEFT=1,CHANGE_LANE_RIGHT=2,JOIN_PLATOON_AT_REAR=3,PLATOON_FOLLOWER_JOIN=4};
```

Comment:
Is there not an enum type already defined for this? Seems weird to define this type in the parsing method and only in the parsing method. Is this enum part of the ROS message structure that could be referenced or something?

Reply:
Yes, you're right, it's already defined. This is not needed.